plugin/pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -0,0 +1,69 @@
+/*
```

Comment:
hmmm... all of the changes to checkServiceAffinity are dependent on precomputation working properly though :) .   without it, checkServiceAffinity will still compile - but it wont ever have a complete pod affinity cache, so the match will  return true for all nodes, because it wont see any affinity constraints, and then the unit tests would fail..